### PR TITLE
shfmt: update to 3.5.1.

### PR DESCRIPTION
**Pull request by mhmdanas**
#### Testing the changes
- I tested the changes in this PR: **YES**

### CODE REVIEW

File: srcpkgs/shfmt/template [hunk 1]
```diff
@@ -14,10 +14,14 @@ license="BSD-3-Clause"
...
+checksum=f15ca9952ef6dc4c1051550152768a99dde0d3f72269d0510f75522a492270cf
 
 post_install() {
 	scdoc < cmd/shfmt/shfmt.1.scd > shfmt.1
 	vman shfmt.1
 	vlicense LICENSE
 }
+
+do_check() {
+	go test ./...
```
Comment (Piraty) [inline srcpkgs/shfmt/template:26]: tests fail for me locally

**Review by Piraty — CHANGES_REQUESTED** [review]

File: srcpkgs/shfmt/template (hunk 1 above)
Comment (mhmdanas) [reply srcpkgs/shfmt/template:26]: Huh, weird, works fine both locally and on the CI. Shall I remove it?

Comment (Piraty) [reply srcpkgs/shfmt/template:26]: in a fresh hostdir:
```
=> shfmt-3.5.1_1: running do_check ...
go: downloading github.com/rogpeppe/go-internal v1.8.1
go: downloading golang.org/x/sync v0.0.0-20210220032951-036812b2e83c
go: downloading github.com/creack/pty v1.1.17
go: downloading github.com/frankban/quicktest v1.14.0
 go: downloading github.com/google/go-cmp v0.5.6
go: downloading gopkg.in/errgo.v2 v2.1.0
go: downloading github.com/kr/pretty v0.3.0
go: downloading github.com/kr/text v0.2.0
# mvdan.cc/sh/v3/shell.test
loadelf: $WORK/b089/_pkg_.a(_x001.o): 129398: sym#52 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b089/_pkg_.a(_x002.o): 129402: sym#52 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b089/_pkg_.a(_x003.o): 129411: sym#70 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b089/_pkg_.a(_x004.o): 129417: sym#57 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x001.o): 129421: sym#50 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x002.o): 129425: sym#27 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x003.o): 129430: sym#64 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x004.o): 129437: sym#65 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x005.o): 129448: sym#82 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x006.o): 129458: sym#97 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x007.o): 129463: sym#68 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x008.o): 129468: sym#64 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x009.o): 129473: sym#83 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x010.o): 129478: sym#64 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x011.o): 129486: sym#67 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x012.o): 129491: sym#75 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
_cgo_callers: relocation target x_cgo_callers not defined
_cgo_init: relocation target x_cgo_init not defined
_cgo_mmap: relocation target x_cgo_mmap not defined
_cgo_munmap: relocation target x_cgo_munmap not defined
_cgo_notify_runtime_init_done: relocation target x_cgo_notify_runtime_init_done not defined
/usr/lib/go/pkg/tool/linux_amd64/link: too many errors
# mvdan.cc/sh/v3/expand.test
loadelf: $WORK/b089/_pkg_.a(_x001.o): 129429: sym#52 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b089/_pkg_.a(_x002.o): 129433: sym#52 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b089/_pkg_.a(_x003.o): 129442: sym#70 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b089/_pkg_.a(_x004.o): 129448: sym#57 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x001.o): 129452: sym#50 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x002.o): 129456: sym#27 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x003.o): 129461: sym#64 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x004.o): 129468: sym#65 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x005.o): 129479: sym#82 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x006.o): 129489: sym#97 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x007.o): 129494: sym#68 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x008.o): 129499: sym#64 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x009.o): 129504: sym#83 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x010.o): 129509: sym#64 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x011.o): 129517: sym#67 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x012.o): 129522: sym#75 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
_cgo_callers: relocation target x_cgo_callers not defined
_cgo_init: relocation target x_cgo_init not defined
_cgo_mmap: relocation target x_cgo_mmap not defined
_cgo_munmap: relocation target x_cgo_munmap not defined
_cgo_notify_runtime_init_done: relocation target x_cgo_notify_runtime_init_done not defined
/usr/lib/go/pkg/tool/linux_amd64/link: too many errors
# mvdan.cc/sh/v3/cmd/gosh.test
loadelf: $WORK/b089/_pkg_.a(_x001.o): 139926: sym#52 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b089/_pkg_.a(_x002.o): 139930: sym#52 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b089/_pkg_.a(_x003.o): 139939: sym#70 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b089/_pkg_.a(_x004.o): 139945: sym#57 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x001.o): 139949: sym#50 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x002.o): 139953: sym#27 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x003.o): 139958: sym#64 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x004.o): 139965: sym#65 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x005.o): 139976: sym#82 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x006.o): 139986: sym#97 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x007.o): 139991: sym#68 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x008.o): 139996: sym#64 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x009.o): 140001: sym#83 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x010.o): 140006: sym#64 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x011.o): 140014: sym#67 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x012.o): 140019: sym#75 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
_cgo_callers: relocation target x_cgo_callers not defined
_cgo_init: relocation target x_cgo_init not defined
_cgo_mmap: relocation target x_cgo_mmap not defined
_cgo_munmap: relocation target x_cgo_munmap not defined
_cgo_notify_runtime_init_done: relocation target x_cgo_notify_runtime_init_done not defined
/usr/lib/go/pkg/tool/linux_amd64/link: too many errors
FAIL	mvdan.cc/sh/v3/cmd/gosh [build failed]
ok  	mvdan.cc/sh/v3/cmd/shfmt	1.402s
FAIL	mvdan.cc/sh/v3/expand [build failed]
?   	mvdan.cc/sh/v3/fileutil	[no test files]
# mvdan.cc/sh/v3/interp.test
loadelf: $WORK/b089/_pkg_.a(_x001.o): 143163: sym#52 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b089/_pkg_.a(_x002.o): 143167: sym#52 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b089/_pkg_.a(_x003.o): 143176: sym#70 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b089/_pkg_.a(_x004.o): 143182: sym#57 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x001.o): 143186: sym#50 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x002.o): 143190: sym#27 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x003.o): 143195: sym#64 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x004.o): 143202: sym#65 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x005.o): 143213: sym#82 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x006.o): 143223: sym#97 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x007.o): 143228: sym#68 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x008.o): 143233: sym#64 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x009.o): 143238: sym#83 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x010.o): 143243: sym#64 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x011.o): 143251: sym#67 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
loadelf: $WORK/b090/_pkg_.a(_x012.o): 143256: sym#75 (wm4.0.fc9c9d2fb651cef3b6c70ac97a70a303): ignoring symbol in section 1 (type 0)
_cgo_callers: relocation target x_cgo_callers not defined
_cgo_init: relocation target x_cgo_init not defined
_cgo_mmap: relocation target x_cgo_mmap not defined
_cgo_munmap: relocation target x_cgo_munmap not defined
_cgo_notify_runtime_init_done: relocation target x_cgo_notify_runtime_init_done not defined
/usr/lib/go/pkg/tool/linux_amd64/link: too many errors
FAIL	mvdan.cc/sh/v3/interp [build failed]
ok  	mvdan.cc/sh/v3/pattern	0.008s
FAIL	mvdan.cc/sh/v3/shell [build failed]
ok  	mvdan.cc/sh/v3/syntax	3.761s
FAIL
=> ERROR: shfmt-3.5.1_1: do_check: 'go test ./...' exited with 2
=> ERROR:   in do_check() at srcpkgs/shfmt/template:26
```

**Comment by ericonr** [comment]
musl:

```
=> shfmt-3.5.1_1: running do_check ...
go: downloading github.com/rogpeppe/go-internal v1.8.1
go: downloading github.com/frankban/quicktest v1.14.0
go: downloading github.com/google/go-cmp v0.5.6
go: downloading golang.org/x/sync v0.0.0-20210220032951-036812b2e83c
go: downloading github.com/creack/pty v1.1.17
go: downloading gopkg.in/errgo.v2 v2.1.0
go: downloading github.com/kr/pretty v0.3.0
go: downloading github.com/kr/text v0.2.0
ok      mvdan.cc/sh/v3/cmd/gosh 0.006s
ok      mvdan.cc/sh/v3/cmd/shfmt        0.060s
ok      mvdan.cc/sh/v3/expand   0.003s
?       mvdan.cc/sh/v3/fileutil [no test files]
ok      mvdan.cc/sh/v3/interp   0.253s
ok      mvdan.cc/sh/v3/pattern  0.003s
ok      mvdan.cc/sh/v3/shell    0.008s
ok      mvdan.cc/sh/v3/syntax   0.662s
```

glibc:

```
=> shfmt-3.5.1_1: running do_check ...
go: downloading github.com/rogpeppe/go-internal v1.8.1
go: downloading github.com/frankban/quicktest v1.14.0
go: downloading github.com/google/go-cmp v0.5.6
go: downloading golang.org/x/sync v0.0.0-20210220032951-036812b2e83c
go: downloading github.com/creack/pty v1.1.17
go: downloading github.com/kr/pretty v0.3.0
go: downloading github.com/kr/text v0.2.0
go: downloading gopkg.in/errgo.v2 v2.1.0
ok      mvdan.cc/sh/v3/cmd/gosh 0.016s
ok      mvdan.cc/sh/v3/cmd/shfmt        0.059s
ok      mvdan.cc/sh/v3/expand   0.003s
?       mvdan.cc/sh/v3/fileutil [no test files]
ok      mvdan.cc/sh/v3/interp   0.488s
ok      mvdan.cc/sh/v3/pattern  0.002s
ok      mvdan.cc/sh/v3/shell    0.004s
ok      mvdan.cc/sh/v3/syntax   0.906s
```

**Comment by mhmdanas** [comment]
So, what now? Keep the test command or remove it?

**Comment by Piraty** [comment]
shrug

**Pull request merged**